src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -158,6 +181,795 @@ MultiLogicalPlanCreate(Query *originalQuery, Query *queryTree)
 
 
 /*
+ * MultiSubqueryPlanTree gets the query objects and returns logical plan
+ * for subqueries.
+ *
+ * We currently have two different code paths for creating logic plan for subqueries:
+ *   (i) subquery pushdown
+ *   (ii) single relation repartition subquery
+ *
+ * In order to do create the logical plan, we follow the algorithm below:
+ *    -  If subquery pushdown planner can plan the query
+ *        -  We're done, we create the multi plan tree and return
+ *    -  Else
+ *       - If the query is not eligible for single table repartition subquery planning
+ *            - Throw the error that the subquery pushdown planner generated
+ *       - If it is eligible for single table repartition subquery planning
+ *            - Check for the errors for single table repartition subquery planning
+ *                - If no errors found, we're done. Create the multi plan and return
+ *                - If found errors, throw it
```

Comment:
> The reverse is also correct. Someone who is trying to run subquery pushdown query with a single table will see errors related repartition subquery planning.

That seems much less frequent to me. Because considering our use cases, if you're to write queries for subquery pushdown, you'd very likely to have multiple `RTE_RELATION`s in the queries. In that case, we'd error out using subquery pushdown errors. For example, all the queries we have in our production customers and `Raw Queries` document have multiple `RTE_RELATION`s. What do you think?

Also, I'm really not comfortable with removing all error messages for repartition subqueries.

The main source of the problem is that we have two separate planners for subqueries, but, we cannot fix it now 😄